tests/queries/0_stateless/01956_fuse_quantile_optimization.sql

```diff
@@ -0,0 +1,54 @@
+DROP TABLE IF EXISTS datetime;
+CREATE TABLE datetime (d DateTime('UTC')) ENGINE = Memory;
+INSERT INTO datetime(d) VALUES(toDateTime('2016-06-15 23:00:00', 'UTC'))
+
+SELECT quantile(0.2)(d), quantile(0.3)(d) FROM datetime;
+SELECT quantileDeterministic(0.2)(d, 1), quantileDeterministic(0.5)(d, 1) FROM datetime;
+SELECT quantileExact(0.2)(d), quantileExact(0.5)(d) FROM datetime;
+SELECT quantileExactWeighted(0.2)(d, 1), quantileExactWeighted(0.4)(d, 1), quantileExactWeighted(0.3)(d, 2) FROM datetime;
+SELECT quantileTiming(0.2)(d), quantileTiming(0.3)(d), quantileTiming(0.2)(d+1) FROM datetime;
+SELECT quantileTimingWeighted(0.2)(d, 1), quantileTimingWeighted(0.3)(d, 1), quantileTimingWeighted(0.2)(d, 2) FROM datetime;
+SELECT quantileTDigest(0.2)(d), quantileTDigest(0.3)(d), quantileTDigest(0.2)(d + 1) FROM datetime;
+SELECT quantileTDigestWeighted(0.2)(d, 1), quantileTDigestWeighted(0.3)(d, 1), quantileTDigestWeighted(0.4)(d, 2) FROM datetime;
+SELECT quantileBFloat16(0.2)(d), quantileBFloat16(0.3)(d), quantileBFloat16(0.4)(d + 1) FROM datetime;
+
+
+SELECT '---------After fuse result-----------';
+set optimize_fuse_quantile=true;
+SELECT 'quantile:';
+EXPLAIN SYNTAX SELECT quantile(0.2)(d), quantile(0.3)(d) FROM datetime;
```

Comment:
Why do we got two `AS y` in the result in `(quantiles(0.5, 0.9, 0.95)(x) AS y)[1] AS y`? 
```
EXPLAIN SYNTAX
SELECT
    b,
    quantile(0.5)(x) AS y,
    quantile(0.9)(x),
    quantile(0.95)(x)
FROM
(
    SELECT
        number AS x,
        number % 2 AS b
    FROM numbers(10)
)
GROUP BY b;

┌─explain──────────────────────────────────────────┐
│ SELECT                                           │
│     b,                                           │
│     (quantiles(0.5, 0.9, 0.95)(x) AS y)[1] AS y, │
│     y[2],                                        │
│     y[3]                                         │
│ FROM                                             │
│ (                                                │
│     SELECT                                       │
│         number AS x,                             │
│         number % 2 AS b                          │
│     FROM numbers(10)                             │
│ )                                                │
│ GROUP BY b                                       │
└──────────────────────────────────────────────────┘
```

Reply:
fixed.